### PR TITLE
e2e: CircleCI housekeeping

### PR DESCRIPTION
**Pull request by Phrynobatrachus**
I added the license upload success/fail check back in now that the server issue there is fixed, and also updated legacy convenience images to new versions as [recommended by Circle](https://discuss.circleci.com/t/legacy-convenience-image-deprecation/41034).

### CODE REVIEW

**Comment by Phrynobatrachus** [comment]
Yeah I'm not sure what's up with that, I'm re-running it to see if it'll complete this time. The postgres logs for the Cypress jobs looked normal otherwise, which should be the only thing affected on those particular ones.

File: .circleci/config.yml [hunk 1]
```diff
@@ -308,12 +309,12 @@ jobs:
   test-MySQL-Postgres-latest:
```
Comment (lieut-data) [inline .circleci/config.yml:309]: Seems like this target was exclusively for `-latest`? Is there an equivalent for `cimg`?

File: .circleci/config.yml [hunk 2]
```diff
@@ -290,11 +291,11 @@ jobs:
   test-MariaDB-Postgres12:
     docker:
       - image: docker.io/mattermost/builder:go-1.16.5-node-16.4.0
-      - image: circleci/postgres:12-alpine
+      - image: cimg/postgres:12.9
         environment:
           POSTGRES_USER: mmuser
           POSTGRES_DB: mattermost_test
-      - image: circleci/mariadb:latest
+      - image: cimg/mariadb:10.6
```
Comment (lieut-data) [inline .circleci/config.yml:298]: I like the idea of pinning MariaDB here. Should we rename the target as `test-MariaDB10-Postgres12`?

File: .circleci/config.yml [hunk 3]
```diff
@@ -272,11 +273,11 @@ jobs:
   test-MySQL8-Postgres11:
     docker:
       - image: docker.io/mattermost/builder:go-1.16.5-node-16.4.0
-      - image: circleci/postgres:11-alpine
+      - image: cimg/postgres:11.13
```
Comment (lieut-data) [inline .circleci/config.yml:276]: Is there a way to avoid the minimum version specification, and just target the latest `11.x`? (here, and elsewhere)

**Review by lieut-data — CHANGES_REQUESTED** [review]
Thanks, @Phrynobatrachus! Just a few thoughts on image version selection below.

File: .circleci/config.yml (hunk 1 above)
Comment (Phrynobatrachus) [reply .circleci/config.yml:309]: They appear to have gotten rid of latest tagging in the new batch of images.

**Comment by lieut-data** [comment]
It might be in GitHub, yes. @crspeller, do you have admin access here? Can you take a look?

**Comment by crspeller** [comment]
@lieut-data @Phrynobatrachus I removed that one as a required check.

**Review by lieut-data — APPROVED** [review]
Thanks, @Phrynobatrachus!

**Pull request merged**